lib/phoenix_live_view/channel.ex

```diff
@@ -171,6 +165,29 @@ defmodule Phoenix.LiveView.Channel do
     end
   end
 
+  defp view_handle_event(%{socket: socket} = state, "lv:clear-flash", val, ref) do
+    new_socket =
+      case val do
+        %{"key" => key} -> Utils.clear_flash(socket, key)
+        _ -> Utils.clear_flash(socket)
+      end
+
+    handle_result({:noreply, new_socket}, {:handle_event, 3, ref}, state)
+  end
+
+  defp view_handle_event(%{socket: _socket} = _state, "lv:" <> _ = bad_event, _val, _ref) do
+    raise ArgumentError, """
+    received unknown LiveView event #{inspect(bad_event)}.
+    The follow LiveView events are suppported: lv:clear-flash.
+    """
+  end
+
+  defp view_handle_event(%{socket: socket} = state, event, val, ref) do
+    event
+    |> socket.view.handle_event(val, socket)
+    |> handle_result({:handle_event, 3, ref}, state)
```

Comment:
Move the handle_result call back to [here](https://github.com/phoenixframework/phoenix_live_view/pull/625/files#diff-4d1a38dc922485fd27a4c315b5c3d489R87) so you don't need to duplicate it on all `lv:` events?

Something like:

    view_handle_event(state.socket, event, val)
    |> handle_result({:handle_event, 3, msg.ref}, state)

Reply:
ah, beautiful !